test_model_configs/single_population_node_with_updaters_config.txt

```diff
@@ -0,0 +1,732 @@
+                      sub_filters {
+                         name: "email_user_info.demo.age_from"
+                         op: EQUAL
+                         value: "18"
+                      }
+                      sub_filters {
+                         name: "email_user_info.demo.age_to"
+                         op: EQUAL
+                         value: "24"
+                      }
+                   }
+                   update {
+                      device_user_type: "email"
+                      corrected_demo {
+                        gender: 1
+                        age {
+                          min_age: 18
+                          max_age: 24
+                        }
+                      }
+                   }
+                }
+                nodes {
+                   condition {
+                      op: AND
+                      sub_filters {
+                         name: "email_user_info.demo.gender"
+                         op: EQUAL
+                         value: "1"
+                      }
+                      sub_filters {
+                         name: "email_user_info.demo.age_from"
+                         op: EQUAL
+                         value: "25"
+                      }
+                      sub_filters {
+                         name: "email_user_info.demo.age_to"
+                         op: EQUAL
+                         value: "34"
+                      }
+                   }
+                   update {
+                      device_user_type: "email"
+                      corrected_demo {
+                        gender: 1
+                        age {
+                          min_age: 25
+                          max_age: 34
+                        }
+                      }
+                   }
+                }
+                nodes {
+                   condition {
+                      op: AND
+                      sub_filters {
+                         name: "email_user_info.demo.gender"
+                         op: EQUAL
+                         value: "1"
+                      }
+                      sub_filters {
+                         name: "email_user_info.demo.age_from"
+                         op: EQUAL
+                         value: "35"
+                      }
+                      sub_filters {
+                         name: "email_user_info.demo.age_to"
+                         op: EQUAL
+                         value: "99"
+                      }
+                   }
+                   update {
+                      device_user_type: "email"
+                      corrected_demo {
+                        gender: 1
+                        age {
+                          min_age: 35
+                          max_age: 99
+                        }
+                      }
+                   }
+                }
+              }
+            }
+            updates{
+              update_matrix {
```

Comment:
update_matrix need a random_seed.

Reply:
<img width="711" alt="Screen Shot 2021-06-03 at 1 53 28 PM" src="https://user-images.githubusercontent.com/65198074/120690335-13f46900-c473-11eb-8c4a-3c5f46219d32.png">

In the code, the random_seed was optional. Would this cause an error if we don't have it?

Reply:
All singular fields are marked as "optional", this is just for proto backwards compatiability. It doesn't mean that it is optional when use this proto. For UpdateMatrix, random_seed is required; it is used to choose a row. And yes, it will cause an error when the labeler loads the model proto.